llama-index-integrations/readers/llama-index-readers-document360/llama_index/readers/document360/base.py

```diff
@@ -0,0 +1,307 @@
+import requests
+from tenacity import retry, stop_after_attempt, wait_fixed, retry_if_exception_type
+
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.schema import Document
+from .entities import Article, ArticleSlim, Category, ProjectVersion
+from .errors import RateLimitException
+
+BASE_URL = "https://apihub.document360.io/v2"
+
+
+class Document360Reader(BaseReader):
+    def __init__(
+        self,
+        api_key: str,
+        should_process_project_version=None,
```

Comment:
can we add types to these arguments? I have no idea what I should be passing for these items

Reply:
Updated